src/mapping/impl/BasisFunctions.hpp

```diff
@@ -205,16 +334,36 @@ class CompactThinPlateSplinesC2 : public CompactSupportBase,
 
   double evaluate(double radius) const
   {
-    double const p = radius * _r_inv;
-    if (p >= 1)
+    return this->operator()(radius, this->_params);
+  }
+
+  PRECICE_HOST_DEVICE inline double operator()(const double radius, const std::array<double, 3> &params) const
+  {
+    double       r_inv = params.at(0);
+    double const p     = radius * r_inv;
+    if (p >= 1) {
       return 0.0;
-    return 1.0 - 30.0 * math::pow_int<2>(p) - 10.0 * math::pow_int<3>(p) + 45.0 * math::pow_int<4>(p) - 6.0 * math::pow_int<5>(p) - math::pow_int<3>(p) * 60.0 * std::log(std::max(p, math::NUMERICAL_ZERO_DIFFERENCE));
+    } else {
+      return 1.0 - 30.0 * std::pow(p, 2) - 10.0 * std::pow(p, 3) + 45.0 * std::pow(p, 4) - 6.0 * std::pow(p, 5) - std::pow(p, 3) * 60.0 * std::log(std::max(p, NUMERICAL_ZERO_DIFFERENCE));
```

Comment:
```suggestion
      return 1.0 - 30.0 * math::pow_int(p, 2) - 10.0 * std::pow(p, 3) + 45.0 * std::pow(p, 4) - 6.0 * std::pow(p, 5) - std::pow(p, 3) * 60.0 * std::log(std::max(p, NUMERICAL_ZERO_DIFFERENCE));
```
similar the others..